.github/workflows/ai-pr-review.yml

```diff
@@ -0,0 +1,149 @@
+            echo "is_workflow_dispatch=true" >> $GITHUB_OUTPUT
+          else
+            echo "is_workflow_dispatch=false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Check branch name
+        id: check-branch
+        if: steps.trigger-type.outputs.is_workflow_dispatch != 'true'
+        env:
+          HEAD_REF: ${{ github.event.pull_request.head.ref }}
+        run: |
+          echo "Branch name: $HEAD_REF"
+          if [[ "$HEAD_REF" == deps/* ]]; then
+            echo "❌ Branch starts with deps/"
+            echo "is_deps_branch=true" >> $GITHUB_OUTPUT
+          else
+            echo "✅ Branch does not start with deps/"
+            echo "is_deps_branch=false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Check if review is requested for hashdotai
+        id: check-reviewer
+        if: steps.trigger-type.outputs.is_workflow_dispatch != 'true'
+        env:
+          REVIEWER_LOGIN: ${{ github.event.requested_reviewer.login }}
+        run: |
+          echo "Requested reviewer: $REVIEWER_LOGIN"
+          if [[ "$REVIEWER_LOGIN" == "hashdotai" ]]; then
+            echo "✅ Review requested for hashdotai"
+            echo "is_hashdotai=true" >> $GITHUB_OUTPUT
+          else
+            echo "❌ Review not requested for hashdotai"
+            echo "is_hashdotai=false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Check PR author
+        id: check-maintainer
+        if: steps.trigger-type.outputs.is_workflow_dispatch != 'true' && steps.check-reviewer.outputs.is_hashdotai == 'true'
+        env:
+          PR_AUTHOR: ${{ github.event.pull_request.user.login }}
+          REPO: ${{ github.repository }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          echo "Checking if $PR_AUTHOR is a maintainer on $REPO"
+          
+          # Get the permission level of the PR author
+          PERMISSION=$(curl -s -H "Authorization: token $GITHUB_TOKEN" \
+            "https://api.github.com/repos/$REPO/collaborators/$PR_AUTHOR/permission" | \
+            jq -r '.permission')
+          
+          # Check if the permission level indicates maintainer status (admin or write)
+          if [[ "$PERMISSION" == "admin" || "$PERMISSION" == "write" ]]; then
+            echo "✅ $PR_AUTHOR is a maintainer"
+            echo "is_maintainer=true" >> $GITHUB_OUTPUT
+          else
+            echo "❌ $PR_AUTHOR is not a maintainer"
+            echo "is_maintainer=false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Set final output
+        id: set-output
+        env:
+          IS_WORKFLOW_DISPATCH: ${{ steps.trigger-type.outputs.is_workflow_dispatch }}
+          IS_HASHDOTAI: ${{ steps.check-reviewer.outputs.is_hashdotai }}
+          IS_MAINTAINER: ${{ steps.check-maintainer.outputs.is_maintainer }}
+          IS_DEPS_BRANCH: ${{ steps.check-branch.outputs.is_deps_branch }}
+        run: |
+          if [[ "$IS_WORKFLOW_DISPATCH" == "true" ]]; then
```

Comment:
Similarly, you can also directly use the `${{ steps. ... == "true" }}` syntax here.

Not really an issue but worth flagging that this is possible.